File: src/main/java/com/github/liebharc/queryenrichment/SelectorBuilder.java
package com.github.liebharc.queryenrichment;

/**
 * Helper class to make it easier to build selectors.
 * @param <T> Attribute type.
 */
public class SelectorBuilder<T> {
    /** The attribute for the selector */
    private Attribute<T> attribute;
    /** The column or property name */
    private String column;

    public SelectorBuilder(Attribute<T> attribute) {
        this.attribute = attribute;
        this.column = attribute.getProperty();
    }

    public SelectorBuilder<T> addColumn(String column) {
        this.column = column;
        return this;
    }

    public Step<T> build() {
        return new Selector<>(attribute, column);
    }
}

File: src/main/java/com/github/liebharc/queryenrichment/SelectorBuilder.kt
package com.github.liebharc.queryenrichment

/**
 * Helper class to make it easier to build selectors.
 * @param <T> Attribute type.
</T> */
class SelectorBuilder<T>(
        /** The attribute for the selector  */
        private val attribute: Attribute<T>) {
    /** The column or property name  */
    private var column: String? = null

    init {
        this.column = attribute.property
    }

    fun addColumn(column: String): SelectorBuilder<T> {
        this.column = column
        return this
    }

    fun build(): ExecutableStep<T, Any?> {
        return Selector(attribute, column)
    }
}
